libMBIN/Source/NMS/GameComponents/GcMessageNPCBehaviourEvent.cs

```diff
@@ -13,7 +13,5 @@ public class GcMessageNPCBehaviourEvent : NMSTemplate
         /* 0x20 */ public GcNPCTriggerTypes InteractionTrigger;
         /* 0x24 */ public int InteractionSubType;
         /* 0x30 */ public Vector3f Position;
-        [NMS(Size = 0x10, Ignore = true)]
-        /* 0x40 */ public byte[] EndPadding;  // this is really weird but there doesn't seem to be anything here and the size *is* 0x50...
```

Comment:
These I think need it. As you can see from my comment. The class should be 0x50 bytes, however the data only takes up 0x40 of them... A lot of these message ones seem to have them and I have no idea why but we should keep them. (Same with the GcAtlasSendSubmitSettlement.cs potentially?

Reply:
Yes, this, GcAtlasSendSubmitSettlement and GcMessagePetBehaviourEvent are all required.
Almost certanly just a field that is not serialized in the mxml.